File: api/src/main/java/proyectouno/api/entity/Reserva.java
package proyectouno.api.entity;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "reservas")
@Getter
@Setter
@ToString(exclude = {"usuario", "clase"})
@EqualsAndHashCode(exclude = {"usuario", "clase"})

public class Reserva {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idReserva;
    @ManyToOne
    @JsonIgnoreProperties("reservas")
    @JoinColumn(name = "idUsuario", nullable = false)
    private Usuario usuario;
    @ManyToOne
    @JsonIgnoreProperties("reservas")
    @JoinColumn(name = "idClase", nullable = false)
    private Clase clase;
    @Column(name = "fechaReserva", nullable = false)
    private LocalDate fechaReserva;
    @Column(name = "estado", nullable = false)
    private String estado;

}

File: Proyecto_II/app/Models/Reserva.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Reserva extends Model
{
    protected $table = 'reservas';

    protected $primaryKey = 'idReserva';

    public $timestamps = false;

    protected $fillable = [
        'idUsuario',
        'idClase',
        'fechaReserva',
        'estado'
    ];

    protected $casts = [
        'fechaReserva' => 'date'
    ];

    public function usuario()
    {
        return $this->belongsTo(Usuario::class, 'idUsuario');
    }

    public function clase()
    {
        return $this->belongsTo(Clase::class, 'idClase');
    }
}
